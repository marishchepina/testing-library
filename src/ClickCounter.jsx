import React, { useCallback, useState } from "react";

const ClickCounter = () => {
  const [counter, setCounter] = useState(0);
  const onClickHandler = useCallback(() => setCounter(counter + 1), [counter]);
  const buttonText = counter ? "Click one more time" : "Make the first click!";

  return (
    <div>
      <button onClick={onClickHandler}>{buttonText}</button>
      <p>{counter}</p>
    </div>
  );
};

export default ClickCounter;
