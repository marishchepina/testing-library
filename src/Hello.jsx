import React from "react";

const Hello = ({ name }) => <h1>Hello {name || "User"}!</h1>;

export default Hello;
