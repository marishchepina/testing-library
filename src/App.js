import Hello from "./Hello";
import ClickCounter from "./ClickCounter";
import OpenClose from "./OpenClose";
import "./styles.css";

const text =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";

const App = () => (
  <div>
    <Hello name="there" />
    <ClickCounter />
    <OpenClose text={text} />
  </div>
);

export default App;
