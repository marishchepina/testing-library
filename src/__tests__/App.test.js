import React from "react";
import { render } from "@testing-library/react";
import App from "../App";
import ClickCounter from "../ClickCounter";
import Hello from "../Hello";

describe("App component tests", () => {
  it("renders App component", () => {
    render(<App />);
    // screen.debug();
  });

  it("should contains Hello and ClickCounter", () => {
    render(<Hello />);
    render(<ClickCounter />);
  });

  it("it match  snapshot", () => {
    //I created snapshot just for curiosity.
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
    // screen.debug();
  });

  it("renders text in App component", () => {
    const { getByText } = render(<App />);
    const text = getByText(
      /Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum./i
    );
    expect(text).toBeInTheDocument();
  });
});
