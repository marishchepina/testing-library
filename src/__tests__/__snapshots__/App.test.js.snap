// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it match  snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello there!
    </h1>
    <div>
      <button>
        Make the first click!
      </button>
      <p>
        0
      </p>
    </div>
    <div>
      <button>
        Hide text
      </button>
      <p
        class="shown"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component tests it match  snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello there!
    </h1>
    <div>
      <button>
        Make the first click!
      </button>
      <p>
        0
      </p>
    </div>
    <div>
      <button>
        Hide text
      </button>
      <p
        class="shown"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component tests renders App component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello there!
    </h1>
    <div>
      <button>
        Make the first click!
      </button>
      <p>
        0
      </p>
    </div>
    <div>
      <button>
        Hide text
      </button>
      <p
        class="shown"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component tests renders App component 2`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello there!
    </h1>
    <div>
      <button>
        Make the first click!
      </button>
      <p>
        0
      </p>
    </div>
    <div>
      <button>
        Hide text
      </button>
      <p
        class="shown"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
